shared/message/motion/KickCommand.proto

```diff
@@ -23,42 +23,80 @@ package message.motion;
 import "Vector.proto";
 
 /**
- * TODO document
- *
- * @author Trent Houliston
- * @author Brendan Annable
+ * Type is NORMAL or POWER (POWER means disregard robot safety and smash it).
  */
 enum KickCommandType {
     NORMAL = 0;
     POWER  = 1;
 }
 
+/**
+ * Contains kick target, direction, and magnitude.
+ */
 message KickCommand {
-    vec3            target          = 1;  // The point to kick
-    vec3            direction       = 2;  // force is the magnitude
+
+    /**
+     * The {x, y, z} point to kick.
+     */
+    vec3 target = 1;
+
+    /**
+     * Force is the magnitude.
+     */
+    vec3 direction = 2;
+
+    /**
+     * Type is NORMAL or POWER (POWER means disregard robot safety and smash it).
+     */
     KickCommandType kickCommandType = 3;
 }
 
+
 /**
- * TODO document
- *
- * @author Trent Houliston
- * @author Brendan Annable
+ * Contains kick direction and which leg to kick with.
  */
 message KickScriptCommand {
-    vec3   direction = 1;  // Direction to kick with magnitude determining force
-    uint32 leg       = 2;  // Leg to kick with
+
+    /**
+     * Direction to kick with magnitude determining force.
+     */
+    vec3 direction = 1;
+
+    /**
+     * Leg to kick with. Use LimbID::LEFT_LEG or LimbID::RIGHT_LEG to check value.
+     */
+    uint32 leg = 2;
 }
 
+/**
+ * Contains config info for kick planner.
+ */
 message KickPlannerConfig {
-    float max_ball_distance        = 1;
-    float kick_corridor_width      = 2;
-    float seconds_not_seen_limit   = 3;
+
+    /**
+     * Maximum distance the ball can be away, in metres.
+     */
+    float max_ball_distance = 1;
+
+    float kick_corridor_width = 2;
```

Comment:
I believe this is used to determine how far apart (laterally) the ball center and foot center are allowed to be while still allowing a kick to happen